### PR TITLE
Adding Customisability for floating window title.

### PR DESCRIPTION
**Pull request by PhilippOesch**
Add support to display the window title relative to the current working directory. See #382

### CODE REVIEW

File: lua/oil/util.lua [hunk 1]
```diff
@@ -429,7 +439,7 @@ M.add_title_to_win = function(winid, opts)
         if vim.api.nvim_win_get_buf(winid) ~= winbuf then
           return
         end
-        local new_title = get_title()
+        local new_title = M.get_title()
```
Comment (stevearc) [inline lua/oil/util.lua:442]: Should pass the winid here

File: lua/oil/util.lua [hunk 2]
```diff
@@ -360,28 +360,38 @@ M.is_floating_win = function(winid)
...
+  local title = vim.api.nvim_buf_get_name(src_buf)
+  local scheme, path = M.parse_url(title)
+
+  if config.adapters[scheme] == "files" then
+    assert(path)
+    local fs = require("oil.fs")
+
+    if config.float.get_win_title ~= nil then
+      return config.float.get_win_title(fs.posix_to_os_path(path))
+    end
```
Comment (stevearc) [inline lua/oil/util.lua:377]: I think that to make this more general, the config `get_win_title` function should take a winid and return a string. Then we could move this check-return to the top of the function.

File: lua/oil/config.lua [hunk 3]
```diff
@@ -332,6 +334,7 @@ local M = {}
 ---@field max_height integer
 ---@field border string|string[]
 ---@field win_options table<string, any>
+---@field get_win_title fun(path: string): string
```
Comment (stevearc) [inline lua/oil/config.lua:337]: You'll also need to add an entry on line 347 below. This is for the internal typed representation of the config, below is for the type that is passed in to `oil.setup()`, so is the same except everything is optionally nil.

**Review by stevearc — CHANGES_REQUESTED** [review]
This is a reasonable extension to the configurability. Left a couple notes for small changes.

File: lua/oil/config.lua (hunk 3 above)
Comment (PhilippOesch) [reply lua/oil/config.lua]: Thanks, adjusted.

File: lua/oil/util.lua (hunk 2 above)
Comment (PhilippOesch) [reply lua/oil/util.lua]: Thank you for the response. Implemented the feedback.

File: lua/oil/util.lua (hunk 1 above)
Comment (PhilippOesch) [reply lua/oil/util.lua]: Adjusted.

**Comment by stevearc** [comment]
Looks good, thanks for the PR!

**Pull request merged**